client/extensions/woocommerce/app/reviews/reviews-list.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { range } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import EmptyContent from 'components/empty-content';
+import { fetchReviews } from 'woocommerce/state/sites/reviews/actions';
+import {
+	areReviewsLoading,
+	areReviewsLoaded,
+	getReviews,
+	getTotalReviews
+} from 'woocommerce/state/sites/reviews/selectors';
+import { getReviewsCurrentPage, getReviewsCurrentSearch } from 'woocommerce/state/ui/reviews/selectors';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import Pagination from 'components/pagination';
+import ReviewCard from './review-card';
+import ReviewsFilterNav from './reviews-filter-nav';
+import { updateCurrentReviewsQuery } from 'woocommerce/state/ui/reviews/actions';
+
+class ReviewsList extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		currentStatus: PropTypes.string,
+		currentSearch: PropTypes.string,
+		currentPage: PropTypes.number,
+		reviews: PropTypes.array,
+		reviewsLoading: PropTypes.bool,
+		reviewsLoaded: PropTypes.bool,
+		total: PropTypes.number,
+	};
+
+	componentDidMount() {
+		const { siteId, currentStatus } = this.props;
+		const query = {
+			page: 1,
+			search: '',
+			status: currentStatus,
+		};
+		this.props.updateCurrentReviewsQuery( this.props.siteId, { page: 1, search: '' } );
+		if ( siteId ) {
+			this.props.fetchReviews( siteId, query );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const hasAnythingChanged = (
```

Comment:
Optional. Destructuring props fields that are referenced multiple times would make this less verbose